src/main/java/org/ays/common/util/validation/NoTrailingOrLeadingSpaces.java

```diff
@@ -0,0 +1,39 @@
+package org.ays.common.util.validation;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation to validate a text using {@link NoTrailingOrLeadingSpacesValidator}.
+ */
+@Target(value = ElementType.FIELD)
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = NoTrailingOrLeadingSpacesValidator.class)
+public @interface NoTrailingOrLeadingSpaces {
+
+    /**
+     * Returns the error message when the text is not valid.
+     *
+     * @return the error message
+     */
+    String message() default "must be valid";
```

Comment:
Burada mesaj için de boşluk ile başlayamaz ve bitemez gibi bir şey yazabilirsek güzel olur. Bir önceki hatada vardı bu mesaj